core/types/receipt.go

```diff
@@ -218,3 +219,19 @@ func (r Receipts) ToShardID(i int) uint32 {
 func (r Receipts) MaxToShardID() uint32 {
 	return 0
 }
+
+// FindLogsWithTopic returns all the logs that contain the given receipt
+func FindLogsWithTopic(
+	receipt *Receipt, targetTopic ethcommon.Hash,
+) []*Log {
+	logs := []*Log{}
+	for _, log := range receipt.Logs {
```

Comment:
go through all the logs ,will it have any performance concern on mainnet?

Reply:
Possibly, but this is probably the best we can do since the logs are stored in an array.